File: Random/3.Subarray Sum Equals K.js
/*
    Author : Mohamed Yousef 
    Date   : 2023-01-05
*/
let solve=function(nums,k){
    let defaultDict=new Proxy({},{
        get:(target,name)=>name in target?target[name]:0
        
    })
    defaultDict[0]=1
    let sum=0
    let ans=0
    for(let i of nums){
        sum+=i
        ans+=defaultDict[sum-k]
        defaultDict[sum]+=1
    }
    return ans;
}
console.log(solve([1,1,1],2));

File: Random/3.Subarray Sum Equals K.py
#    Author : Mohamed Yousef 
#    Date   : 2023-01-04
import sys,math,bisect,collections,itertools,heapq
from collections import defaultdict,deque
sys.setrecursionlimit(50000) #To increase Recursion depth in py
def solve(nums,k):
    myDict=defaultdict(int)
    myDict[0]+=1
    sum=0
    ans=0
    for i in nums:
        sum+=i
        store=sum-k
        if store in myDict:
            ans+=myDict[store]
        myDict[sum]+=1
    return ans

print(solve([1,2,3],3))
